lib/common/grpc-service.js

```diff
@@ -446,26 +447,43 @@ GrpcService.createDeadline_ = function(timeout) {
 };
 
 /**
- * Checks for a grpc error code and extends the Error object with additional
+ * Checks for a grpc status code and extends the supplied object with additional
  * information.
  *
  * @private
  *
- * @param {error} err - The original request error.
- * @return {error|null}
+ * @param {object} obj - The object to be extended.
+ * @param {object} response - The grpc response.
+ * @return {object|null}
  */
-GrpcService.getError_ = function(err) {
-  if (err && GRPC_ERROR_CODE_TO_HTTP[err.code]) {
-    var defaultErrorDetails = GRPC_ERROR_CODE_TO_HTTP[err.code];
-    err.code = defaultErrorDetails.code;
-    err.message = err.message || defaultErrorDetails.message;
-    return err;
+GrpcService.extendGrpcResponse_ = function(obj, response) {
```

Comment:
I think this method is a little confusing on its own. It might be best to treat this like a utility method used by `decorateStatus_` and `decorateError_`.

I'm choosing `decorate` as opposed to `get` because `var status = getStatus(status)` isn't really what's happening. We already have the status, we're just elaborating on it. And `extend` is okay, except it could be confused with the extend module in that, "am I supposed to provide an object to extend with?".